crypto/dilithium/p_dilithium3.c

```diff
@@ -119,6 +119,8 @@ const EVP_PKEY_METHOD dilithium3_pkey_meth = {
     NULL /* derive */,
     NULL /* paramgen */,
     NULL /* ctrl */,
+    NULL /* keygen deterministic */,
+    NULL /* encapsulate deterministic */,
     NULL /* encapsulate */,
     NULL /* decapsulate */,
```

Comment:
Weren't we going to get rid of the dedicated function pointers for the randomized versions and handle creating the seed in the EVP layer and calling the deterministic version?

Reply:
It was proposed as a design option (as it works very well for encapsulation), but this method would be very inconsistent with existing library behaviour around `EVP_PKEY_keygen`.

The reason being that `EVP_PKEY_keygen` is generic for all EVP key types (i.e., not KEM specific). The concept of deterministic keygen with a `seed` doesn't apply to all other EVP types than KEMs.

We could get around this by using KEM helper functions such as [EVP_PKEY_CTX_kem_set_params](https://github.com/aws/aws-lc/blob/b6daf74a7cc9a914df4a935af808790361ff30b9/crypto/evp_extra/p_kem.c#L198C2-L221C2), but this seems inconsistent with the traditional use of the EVP API, and could allow mis-use or misconfiguration. In design review this option was not favoured over the implementation in this PR.

Reply:
Follow up from discussion with Andrew and Dusan:

We are in alignment that ideal this behaviour occurs as high up the EVP stack as possible, however, as the current implementation of the de-randomized API in ML-KEM/Kyber is such that the de-randomized mode calls the randomized mode at the algorithm implementation layer (https://github.com/jakemas/aws-lc/blob/35097c28bb96f1b29d064250669682203a762eb2/crypto/kyber/pqcrystals_kyber_ref_common/kem.c#L11-L59) it would be an anti-pattern to design the function pointers in this way.